src/textual/widgets/_button.py

```diff
@@ -1,57 +1,142 @@
 from __future__ import annotations
 
+import sys
 from typing import cast
 
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal  # pragma: no cover
+
 from rich.console import RenderableType
-from rich.style import Style
 from rich.text import Text, TextType
 
 from .. import events
+from ..css._error_tools import friendly_list
 from ..message import Message
 from ..reactive import Reactive
 from ..widget import Widget
 
+ButtonVariant = Literal["default", "success", "warning", "error"]
+_VALID_BUTTON_VARIANTS = {"default", "success", "warning", "error"}
```

Comment:
I hate this :(

Reply:
The repetition? Yeah, feels so awkward.

Reply:
Yep!

Reply:
`_VALID_BUTTON_VARIANTS = ButtonVariant.__args__` would work - and the signature of the Literal class is unlikely to change (it would probably break types checkers).
Not very satisfying for sure, as it feels a bit hacky, but it avoids the repetition :slightly_smiling_face:

Reply:
I think you would get a less efficient tuple there, and I'm not too keen on the use of the dunder attributes.